gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +734,689 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+
+                /* Polar Stereographic / STG projection */
+                } else if( strcmp(pstr,"STG" ) == 0 ) {
+                    projName.assign("Polar_Stereographic_");
+                    oSRS.SetProjection(SRS_PT_POLAR_STEREOGRAPHIC);
+                    oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                    oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+                } else {
+                    CPLError(CE_Failure, CPLE_AppDefined,
+                         "Unknown projection.");
+                }
+
+                projName.append(target);
+                oSRS.SetProjParm(SRS_PP_FALSE_EASTING,0.0);
+                oSRS.SetProjParm(SRS_PP_FALSE_NORTHING,0.0);
+
+                char * cstrproj = new char [projName.length()+1];
+                std::strcpy (cstrproj, projName.c_str());
+                oSRS.SetNode("PROJCS",cstrproj);
+
+                char * cstrgeoname = new char [pszGeogName.length()+1];
+                std::strcpy (cstrgeoname, pszGeogName.c_str());
+                char * cstrdname = new char [pszDatumName.length()+1];
+                std::strcpy (cstrdname, pszDatumName.c_str());
+                oSRS.SetGeogCS(cstrgeoname, cstrdname, target, aRadius, invFlattening,
+                    "Reference_Meridian", 0.0, "degree", 0.0174532925199433);
+            }  else {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                     "Unknown projection.");
+            }
+        }
+    }
+
+    if( m_bLoadPam )
+    {
+/* -------------------------------------------------------------------- */
+/*      Initialize any PAM information.                                 */
+/* -------------------------------------------------------------------- */
+        CPLAssert(!bMetadataChanged);
+        CPLAssert(!bNoDataChanged);
+
+        m_bLoadPam = false;
+        bMetadataChanged = false;
+        bNoDataChanged = false;
+
+        bitpix = this->fitsDataType;
+        FITSRasterBand *poBand = cpl::down_cast<FITSRasterBand*>(GetRasterBand(1));
+
+        if (bitpix != TUSHORT && bitpix != TUINT)
+        {
+            fits_read_key(hFITS, TDOUBLE, "BSCALE", &dfScale, nullptr, &status);
+            if( status )
+            {
+                status = 0;
+                dfScale = 1.;
+            }
+            fits_read_key(hFITS, TDOUBLE, "BZERO", &dfOffset, nullptr, &status);
+            if( status )
+            {
+                status = 0;
+                dfOffset = 0.;
+            }
+        }
+        else {
+            poBand->bHaveOffsetScale = true;
+        }
+
+        if( !poBand->bHaveOffsetScale )
+        {
+            poBand->bHaveOffsetScale = true;
+            poBand->dfScale = dfScale;
+            poBand->dfOffset = dfOffset;
+        }
+
+        fits_read_key(hFITS, TDOUBLE, "BLANK", &dfNoDataValue, nullptr, &status);
+        bNoDataSet = !status;
+
+    m_bLoadPam = false;
```

Comment:
wrong identation

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/f17edf20f02f51decbbc735b0dfe0c4219a20111